pkg/ccl/sqlproxyccl/proxy_handler.go

```diff
@@ -244,19 +261,24 @@ func (handler *proxyHandler) handle(ctx context.Context, proxyConn *conn) error
 				}
 				continue
 			}
-			clientErr := newErrorf(codeParamsRoutingFailed, "cluster %s-%d not found", clusterName, tenID.ToUint64())
-			updateMetricsAndSendErrToClient(clientErr, conn, handler.metrics)
```

Comment:
Why remove this? Don't we want to send err back in this case?

Reply:
We `break` here and call `updateMetricsAndSendErrToClient` metrics below. So it appears we're calling it twice in this codepath. I don't think there's a test for this.